incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -116,17 +116,22 @@ type Condition struct {
 	//
 	// Currently, the supported values are:
 	//
-	// - "CRIT_PARENT_MISSING": the specified parent is missing
+	// - "CritParentMissing": the specified parent is missing
 	//
-	// - "CRIT_PARENT_INVALID": the specified parent is invalid (e.g., would cause a cycle)
+	// - "CritParentInvalid": the specified parent is invalid (e.g., would cause a cycle)
 	//
-	// - "CRIT_ANCESTOR": a critical error exists in an ancestor namespace, so this namespace is no
+	// - "CritAncestor": a critical error exists in an ancestor namespace, so this namespace is no
 	// longer being updated either.
 	//
-	// - "REQUIRED_CHILD_CONFLICT": this namespace has a required child, but a namespace of the same
-	// name already exists and is not a child of this namespace. Note that the condition is _not_
-	// annotated onto the other namespace; it is considered an error _only_ for the would-be parent
-	// namespace.
+	// - "HNSMissing": this namespace is an owned namespace (created by reconciling an hns instance),
+	// but the hns instance is missing in its owner (referenced in its owner annotation) or the owner
+	// namespace is missing.
+	//
+	// - "CannotUpdateObject": this namespace has an error when updating a propagated object. The
+	// affected object will point to the source object.
+	//
+	// - "CannotPropagateObject": this namespace has an error when propagating a source object. The
+	// affected objects will point to all the propagated objects that have an error to update.
```

Comment:
> all the propagated objects that have an error to update.

I am wondering what's the difference between this code and the `CannotUpdateObject` above? From the documentation, they seem to be both about problems when updating propagated objects?

Reply:
Yes, they are two conditions triggered by the same error. However, `CannotPropagateObject` is set on the namespace with the source object and `CannotUpdateObject` is set on the namespace with the propagated object.

Reply:
Maybe: "This object contains an object that couldn't be propagated to one or more of its descendants. The condition's affect objects will include a list of the copies that couldn't be updated."

Reply:
Thanks all, updated!